version: "3.8"
services:
  app:
    container_name: trashbot-app
    image: node:lts
    working_dir: /var/www/html/app/
    entrypoint: /bin/bash
    expose:
      - 8003
    volumes:
      - ./app/dist:/var/www/html/app
    tty: true
    networks:
      - mkl_external
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.devapptrash.rule=Host(`trashbot.app.mobile.media.fhstp.ac.at`)"
      - "traefik.http.routers.devapptrash.entrypoints=web-secure"
      - "traefik.http.routers.devapptrash.tls=true"
      - "traefik.http.routers.devapptrash.tls.certresolver=letsencrypt"
      - "traefik.http.services.devapptrash.loadbalancer.server.port=8003"
  server:
    container_name: trashbot-api
    image: trashbot-api:1.0.0
    build:
      context: .
      dockerfile: ./server/Dockerfile
    restart: unless-stopped
    command: 'npm run run'
    cpus: 3
    networks:
      - mkl_external
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.devapitrash.rule=Host(`trashbot.api.mobile.media.fhstp.ac.at`)'
      - 'traefik.http.routers.devapitrash.entrypoints=web-secure'
      - 'traefik.http.routers.devapitrash.tls=true'
      - 'traefik.http.routers.devapitrash.tls.certresolver=letsencrypt'
      - 'traefik.http.services.devapitrash.loadbalancer.server.port=8000'
networks:
  mkl_external:
    external: true
